models/productDao.js

```diff
@@ -15,15 +13,55 @@ const getProducts = async (offset, limit, gender) => {
        FROM products AS p
        INNER JOIN sub_categories AS sc ON p.sub_category_id = sc.id
        INNER JOIN main_categories AS mc ON sc.main_category_id = mc.id
-       WHERE mc.name = ? AND p.id LIMIT ?,?`, [gender, numberOffset, numberLimit]
+       WHERE mc.name = ? AND p.id LIMIT ?,?`, [gender, offset, limit]
     );
   } catch (err) {
-    const error = new Error("INVALID_DATA_INPUT");
+    const error = new Error(err.message);
     error.statusCode = 500;
     throw error;
   }
 };
 
+const productDetails = async (productId) => {
+   try{
+    return await database.query(`
+      SELECT
+      p.id,
+      p.name,
+      p.price,
+      p.thumbnail_image_url AS thumbnailUrl,
+      mc.name AS gender,
+      sc.name AS cateogory,
+        JSON_ARRAYAGG(pi.image_url) AS images,
+        (	
+          SELECT
+            JSON_ARRAYAGG(
+              JSON_OBJECT(
+                'id', po.id,
+                'footSize', s.foot_size,
+                'stock', po.stock
+            )
+          )
+          FROM product_options AS po
+            INNER JOIN sizes AS s ON s.id = po.size_id
+            WHERE po.product_id = p.id
+        ) AS stocksize
```

Comment:
마이너한 부분이지만 포맷팅 (들여쓰기) 다시 한번 신경써주세요!
```suggestion
      sc.name AS cateogory,
      JSON_ARRAYAGG(pi.image_url) AS images,
      (	
        SELECT
          JSON_ARRAYAGG(
            JSON_OBJECT(
              'id', po.id,
              'footSize', s.foot_size,
              'stock', po.stock
          )
        )
        FROM product_options AS po
          INNER JOIN sizes AS s ON s.id = po.size_id
          WHERE po.product_id = p.id
      ) AS stocksize
```